routes/product/helpers.js

```diff
@@ -89,15 +108,20 @@ export default {
                         { producerId: { $eq: producerId } },
                     ],
                 };
-                data = await Product.updateOne(filter, product);
+
+                await Product.updateOne(filter, product);
+
+                data = product;
+                message = PRODUCT_UPDATED;
             } else {
-                // Redundant code to catch if user insert invalid '_id'
+                // Redundancy: deletes possible invalid/valid '_id'
                 delete product._id;
 
                 data = await Product.create(product);
```

Comment:
how is this possible, passing the new `productData` as param to the instance of the Product model. I need clarifications here

Reply:
So there are many ways (i think about 3 or so) to create a new item using mongoose. This is just want of the ways